bedrock/base/templates/macros.html

```diff
@@ -319,9 +318,14 @@ <h3>{{ _('Your download link was sent.') }}</h3>
       {% do fxa_queries.append('utm_{}={}'.format(param, val.replace('/', '%2F'))) %}
     {% endif %}
   {% endfor %}
+  {% if not service_type %}
+    {% set service_type = 'sync' %}
+  {% endif %}
 
-  href="{{ settings.FXA_ENDPOINT}}{{ fxa_path }}?{{ fxa_queries|join('&') }}" data-mozillaonline-link="{{ settings.FXA_ENDPOINT_MOZILLAONLINE }}{{ fxa_path }}?{{ fxa_queries|join('&') }}"
+  href="{{ settings.FXA_ENDPOINT }}{{ fxa_path }}?service={{ service_type }}&{{ fxa_queries|join('&') }}" data-mozillaonline-link="{{ settings.FXA_ENDPOINT_MOZILLAONLINE }}{{ fxa_path }}?{{ fxa_queries|join('&') }}"
+  class="fxa_link_fragment"
```

Comment:
The buttons that use the `fxa_link_fragment` macro already have a `class` attribute defined in their page CSS, so you can't overwrite that here without browsers ingoring it. This seems to be why the node list returned by your `querySelector` is empty.

What we may need here is to create a new `fxa_cta_link` macro, that consumes `fxa_link_fragment` and returns the full button HTML, instead of just the `href`? We already have `fxa_create_account_button`, but this seems to do way more things than we need. All we require is a single button 😕

If we create an `fxa_cta_link` macro, then you could pass a `button_class` parameter to it so that individual pages could still apply their own classes, whilst still retaining your JS hook, like so:

```
class="js-fxa-cta-link {{ button_class }}"
```

Pages could then call the `fxa_cta_link()` macro directly, and pass in the values they need?

Reply:
I think I'm confused how many places this new button would touch? What does `fxa_cta_link` replace? All fxa buttons on bedrock? is it replacing `fxa_link_fragment` in some situations?
`fxa_create_account_button` is complex, but I'm not sure why this wouldn't belong in it?

Reply:
The fxa_cta_link macro would be used everywhere we show the standard button, instead of creating a custom link everywhere like we do now. Let me know if you want to talk through this in person if it doesn’t make sense.

Reply:
@alexgibson maybe we should set up a time to talk because I'm experiencing that any `{{ button_class }}` value passed will overwrite any preexisting classes. So in the case of: `class="js-fxa-cta-link {{ button_class }}"`, `js-fxa-cta-link` will be overwritten.

Reply:
@amy perhaps this might be easier if you can share your code? If you can push what you have here, i’ll take a look

Reply:
@alexgibson I pushed up a rough draft of `fxa_cta_link` macro. I feel a little clumsy on everywhere it touches. Maybe we can organize a time to chat more?